### PR TITLE
feat: Support HEIF, HEIC and AVIF recipe image uploads

### LINKED ISSUE #3393: [BUG] - Support for HEIC images
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

Unable to upload HEIC images. This is image format used by iPhones

image

### Steps to Reproduce

Enter recipe
click `I Made This`
try to upload HEIC image on desktop

When I change Finder filtering to all files and try to upload then in console 500 internal is thrown

```
Request Method:
PUT
api/recipes/timeline/events/2ef4d933-3cd4-4617-957f-eb0ce3a6545c/image
```

### Please provide relevant logs

``` shell
mealie  | INFO:     192.168.176.3:53266 - "PUT /api/recipes/timeline/events/2ef4d933-3cd4-4617-957f-eb0ce3a6545c/image HTTP/1.1" 500 Internal Server Error
mealie  | ERROR:    Exception in ASGI application
mealie  | Traceback (most recent call last):
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 412, in run_asgi
mealie  |     result = await app(  # type: ignore[func-returns-value]
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
mealie  |     return await self.app(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
mealie  |     await super().__call__(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/applications.py", line 123, in __call__
mealie  |     await self.middleware_stack(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
mealie  |     raise exc
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
mealie  |     await self.app(scope, receive, _send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 24, in __call__
mealie  |     await responder(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 44, in __call__
mealie  |     await self.app(scope, receive, self.send_with_gzip)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
mealie  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
mealie  |     raise exc
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
mealie  |     await app(scope, receive, sender)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 758, in __call__
mealie  |     await self.middleware_stack(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 778, in app
mealie  |     await route.handle(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 299, in handle
mealie  |     await self.app(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 79, in app
mealie  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
mealie  |     raise exc
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
mealie  |     await app(scope, receive, sender)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/routing.py", line 74, in app
mealie  |     response = await func(request)
mealie  |   File "/app/mealie/routes/_base/routers.py", line 35, in custom_route_handler
mealie  |     response = await original_route_handler(request)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 278, in app
mealie  |     raw_response = await run_endpoint_function(
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 193, in run_endpoint_function
mealie  |     return await run_in_threadpool(dependant.call, **values)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/starlette/concurrency.py", line 42, in run_in_threadpool
mealie  |     return await anyio.to_thread.run_sync(func, *args)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/anyio/to_thread.py", line 33, in run_sync
mealie  |     return await get_asynclib().run_sync_in_worker_thread(
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
mealie  |     return await future
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 807, in run
mealie  |     result = context.run(func, *args)
mealie  |   File "/app/mealie/routes/recipe/timeline_events.py", line 136, in update_event_image
mealie  |     data_service.write_image(image, extension, event.image_dir)
mealie  |   File "/app/mealie/services/recipe/recipe_data_service.py", line 100, in write_image
mealie  |     self.minifier.minify(image_path)
mealie  |   File "/app/mealie/pkgs/img/minify.py", line 104, in minify
mealie  |     PillowMinifier.to_webp(image_file, org_dest, quality=70)
mealie  |   File "/app/mealie/pkgs/img/minify.py", line 67, in to_webp
mealie  |     img = Image.open(image_file)
mealie  |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/PIL/Image.py", line 3309, in open
mealie  |     raise UnidentifiedImageError(msg)
mealie  | PIL.UnidentifiedImageError: cannot identify image file '/app/data/recipes/bdef5429-b909-4d50-a84a-5c28d9c374b5/images/timeline/2ef4d933-3cd4-4617-957f-eb0ce3a6545c/original.HEIC'
```

### Mealie Version

_No response_

### Deployment

Docker (Linux)

### Additional Deployment Details

_No response_

### PR DESCRIPTION
**Pull request by tba-code**
## What type of PR is this?

- feature

## What this PR does / why we need it:

- Pillow does not natively support HEIF as seen [here](https://github.com/python-pillow/Pillow/issues/2806).
- Apple products use .heic.
- This PR Adds support for HEIF files (.heif, .heic, .avif). via [pi-heif](https://pypi.org/project/pi-heif/).

## Which issue(s) this PR fixes:

Fixes #3393 

## Testing
Manually tested by downloading an heic image from the internet and uploading it to a test recipe in Mealie.

### CODE REVIEW

File: mealie/pkgs/img/minify.py [hunk 1]
```diff
@@ -3,12 +3,15 @@
...
+from pi_heif import register_heif_opener
 from PIL import Image
 
+register_heif_opener()
+
 WEBP = ".webp"
 FORMAT = "WEBP"
 
-IMAGE_EXTENSIONS = {".jpg", ".jpeg", ".png", ".webp"}
+IMAGE_EXTENSIONS = {".jpg", ".jpeg", ".png", ".webp", ".heic", ".avif"}
```
Comment (michael-genson) [inline mealie/pkgs/img/minify.py:15]: Are you sure that `.avif` files work? I tried a couple and got an `UnidentifiedImageError` from PIL. `.heic` seem to work fine.

**Review by michael-genson — CHANGES_REQUESTED** [review]

File: mealie/pkgs/img/minify.py (hunk 1 above)
Comment (michael-genson) [reply mealie/pkgs/img/minify.py:15]: Also confirmed `.heif` files work

**Comment by tba-code** [comment]
Thanks for pointing out my mistake here. I have fixed it and avif support is now properly included. Please try again.

**Review by michael-genson — APPROVED** [review]
Looks good, thanks!

**Pull request merged**